packages/dev/core/src/Rendering/prePassRenderer.ts

```diff
@@ -284,10 +284,16 @@ export class PrePassRenderer {
             type = Constants.TEXTURETYPE_HALF_FLOAT;
         }
 
-        if (type !== Constants.TEXTURETYPE_FLOAT) {
-            for (let i = 0; i < PrePassRenderer.TextureFormats.length; ++i) {
-                if (PrePassRenderer.TextureFormats[i].type === Constants.TEXTURETYPE_FLOAT) {
-                    PrePassRenderer.TextureFormats[Constants.PREPASS_DEPTH_TEXTURE_TYPE].type = type;
+        for (let i = 0; i < PrePassRenderer.TextureFormats.length; ++i) {
+            const format = PrePassRenderer.TextureFormats[i].format;
+            if (PrePassRenderer.TextureFormats[i].type === Constants.TEXTURETYPE_FLOAT) {

```

Comment:
Isnt all of this only changing on TextureFormats changes ? (wonder if we could cache all the conditions)

Reply:
This code is done only once, in the constructor, to account for the capability of the current hardware. I'm not sure what you mean by caching the conditions?